polaris-react/src/styles/shared/_controls.scss

```diff
@@ -1,5 +1,5 @@
 @function control-height() {
-  @return 36px;
+  @return 32px;
```

Comment:
Value ends up being used in `TextField`, `Select` and `ResourceList`. The first two override the value to be `32px` with the feature flag enabled so they are not affected by this change. 

The third, ResourceList, [doesn't override this style](https://github.com/Shopify/polaris/blob/36b7925ca31e0d08767ac4f1ecb6ac4fccbe3f63/polaris-react/src/components/ResourceList/ResourceList.scss#L89). However it appears its being used as a min-height value, and (from what I can see) doesn't result in any visual change to the ResourceList component. 

Before: 
![Screenshot 2023-08-16 at 5 27 55 pm](https://github.com/Shopify/polaris/assets/12119389/a138ff7a-fa76-41ec-974a-6728da6c32ca)

After: 
![Screenshot 2023-08-16 at 5 28 01 pm](https://github.com/Shopify/polaris/assets/12119389/a906c22d-2b7d-47db-b141-84992ab45320)


Additionally, insofar as the logic described in the following issue around [variable heights](https://github.com/Shopify/polaris-internal/issues/299). It would appear that Select leverages `line-height` and `font-size` to enforce a height of `32px` on larger screens and `36px` on smaller ones. So defaulting the `min-height` specified in the `control-height` mixin to the minimum of `32px` appears to be the right thing to do. At least until we address / resolve the behaviour outlined in the above linked issue. (cc @jesstelford)